### DISCUSSION TITLE
Filter weather alerts by using SAME codes?

### DISCUSSION THREAD

**Discussion by dpoisson2017** [egzumer/uv-k5-firmware-custom#203] [Ideas]
Is it possible to have the weather alerts filtered by SAME codes (up to 3 or 5)?
Weather stations broadcast on a vast territory and alerts are often only for a specific county.

From https://www.weather.gov/iln/nwr_same:
At the beginning of the product transmission, 3 bursts of digital static are heard (which contains 6 digit SAME code, amongst other things)
Followed by 10 second 1050hz alarm tone
Followed by the actual audio message.

Has anyone ever looked into this? Would it even be feasable...I have doubts because of the "digital static" part possibly couldn't be demodulated? To my knowledge, this would make the Quansheng radios the only ham radios to have this functionality.

**Comment by ErikS-web** [comment]
NOAA is standard NOT enabled in this firmware.

There is no use for it in Europa.

See [README.md](https://github.com/egzumer/uv-k5-firmware-custom/blob/main/README.md)
ENABLE_NOAA                   := 0       everything NOAA (only of any use in the USA)

**Comment by egzumer** [comment]
From the datasheet it looks like the RF chip supports the NOAA SAME code decoding, but I'm from Europe where there is no NOAA system, I have no motivation to work on that.

**Comment by dpoisson2017** [comment]
The egzumer mods reputation has crossed the big pond! I'm from Québec, Canada and I have access to NOAA broadcasts as well. Haven't even tried modding my radio (it has been wrapped and placed under the Christmas tree). 

I'm a software developper by trade. I usually play with higher level stuff (web backend and the likes), but I'd love to contribute, but I might have lots of questions. 

Just read: https://forum.arduino.cc/t/decode-noaa-weather-radio-same-codes/84505/76 and it seems rather complex. 
Also, when you speak of datasheet, are you talking about the BK4819 datasheet? I took a glance at it and  couldn't find anything specific pertaining to NOAA / SAME decoding.

**Reply by egzumer** [reply]
BK4819 Registers_V1.1.pdf
REG_58<3:1>
010: NOAA SAME RX

[bk4819.zip](https://github.com/egzumer/uv-k5-firmware-custom/files/13604133/bk4819.zip)
